Source/FullScreenPokemon.ts

```diff
@@ -2511,6 +2524,8 @@ module FullScreenPokemon {
                     this.StateHolder.addChange(other.id, "sight", undefined);
                 }
             }
+
+            this.autoSave();
```

Comment:
By visit, do you mean the stack pops out of `animateTrainerBattleStart`and continues on to the line setting the alreadyBattled attribute to false (which is what happens, right?) or `animateCharacterDialogFinish` is called after the battle is completed?
I'm also confused on exactly we are talking about for these comments. I guess I'm missing the point of the first comment.